File: ios/MoodleAssessmentApp/MoodleAssessmentApp/Data/DTO/GradesResponseDTO.swift
import Foundation

struct GradesResponseDTO: Decodable {
    let usergrades: [UserGradeDTO]?
    let gradeitems: [GradeItemDTO]?
}

struct UserGradeDTO: Decodable {
    let courseid: Int?
    let gradeitems: [GradeItemDTO]?
}

File: android/app/src/main/java/com/itcorner/moodleandroid/data/dto/GradesResponseDto.kt
package com.itcorner.moodleandroid.data.dto

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class GradesResponseDto(
    val usergrades: List<UserGradeDto>? = null
)

@JsonClass(generateAdapter = true)
data class UserGradeDto(
    val courseid: Int? = null,
    val gradeitems: List<GradeItemDto>? = null
)

@JsonClass(generateAdapter = true)
data class GradeItemDto(
    val id: Int,
    val itemname: String? = null,
    val gradeformatted: String? = null,
    val percentageformatted: String? = null,
    val graderaw: Double? = null,
    val grademin: Double? = null,
    val grademax: Double? = null
)
